frame/vesting/src/lib.rs

```diff
@@ -199,7 +199,11 @@ pub mod pallet {
 
 		#[cfg(feature = "try-runtime")]
 		fn post_upgrade() -> Result<(), &'static str> {
-			migrations::v1::post_migrate::<T>()
+			if StorageVersion::<T>::get() == Releases::V0 {
+				migrations::v1::post_migrate::<T>()
+			} else {
+				Ok(())
+			}
```

Comment:
This doesn't make sense. Because the `StorageVersion` should be `V1` after applying the migration.

Reply:
Where is it failing actually?

Reply:
https://github.com/paritytech/substrate/blob/master/frame/vesting/src/migrations.rs#L72

Reply:
And when is this failing? Do you already have applied this upgrade?

Reply:
I just tried the command beblow

```
RUST_LOG=runtime=trace,try-runtime::cli=trace,executor=trace 
./target/release/polkadot try-runtime
--execution Native --chain polkadot-dev
on-runtime-upgrade
live --uri ws://localhost:29944 --at 0x83c1812face22aa9d31f6d0033874a88bf5731d88a9ab8ed44cc727a165ea20b
```